### ISSUE TITLE
Vimwiki changes the CWD to the wrong wiki's path when following a wiki-link from a diary page

### ISSUE THREAD

**Issue by jmtd** [vimwiki/vimwiki#1146]
Vimwiki changes the CWD to the wrong wiki's path when following a wiki-link from a diary page
----
Prior to submitting a new issue make sure to complete these steps:

- [X] Post the syntax you are using (default/mediawiki/markdown) **and** your vimwiki settings from your `.vimrc`
- [X] Provide a detailed description of the problem including **steps to reproduce the issue**.
- [X] Include the output of `:VimwikiShowVersion`. 
   * Stable version: 2.5, and
- [X] Checkout the `dev` branch and confirm the issue is present there as well.

```
Os: Linux
Vim: 800
Branch: dev
Revision: 212a66f
Date: 2021-06-11 12:06:43 -0400
```

----

I have 3 vimwikis configured in my vimrc, as follows

```
let g:vimwiki_list = [{
    \ 'path': '$HOME/rh/2021',
    \ 'ext': '.mdwn',
    \ 'auto_diary_index': 1
    \ },{
    \ 'path': '$HOME/Documents/vimwiki',
    \ 'ext': '.mdwn',
    \ 'auto_diary_index': 1
    \ },{
    \ 'path': '$HOME/git/phd/phdwiki',
    \ 'ext': '.mdwn',
    \ 'auto_diary_index': 1
    \ }
    \ ]
let g:vimwiki_key_mappings = { 'all_maps': 0, }
let g:vimwiki_conceallevel = 0
let g:vimwiki_url_maxsave  = 0 
```

```
$ vim
:pwd
/home/jon
:VimwikiMakeDiaryNote
```
A diary page appears. The page is saved in the right place, under `$HOME/rh/2021/diary`. CWD has not been updated yet:
```
:pwd
/home/jon
```
Now I follow a wiki-link in the diary. The page that is loaded is the correct one, from `$HOME/rh/2021`. But the CWD has been changed, and to the wrong wiki:
```
:pwd
/home/jon/git/phd/phdwiki
```
Interestingly if I re-order vimwiki_list, so phdwiki is not last, it makes no difference: the above sequence has the same outcome.

**Comment by brennen** [comment]
One question - I'm assuming you have `g:vimwiki_auto_chdir` set?

**Comment by brennen** [comment]
I made a cursory attempt to reproduce this with the below config and dev branch at 22dcb366; didn't come up with much.

```vim
" a stub vimrc for just running vimwiki on an otherwise stock setup

set nocompatible
set nomodeline
let mapleader = ","

" vundle setup & vundle-managed plugins {{{

  filetype off

  set rtp+=~/.vim/bundle/Vundle.vim/
  call vundle#begin()

  " let Vundle manage Vundle - required:
  Plugin 'VundleVim/Vundle.vim'

  " CUSTOMIZE: vimwikis - for notes, daily logs, etc. {{{

    " main personal wiki
    let wiki = {}
    let wiki.path = '~/notes/vimwiki/'
    let wiki.path_html = '~/notes/html/'
    let wiki.auto_tags = 1
    let wiki.auto_diary_index = 1
    let wiki.automatic_nested_syntaxes = 1

    " a markdown wiki for testing purposes
    let markdownwiki = {}
    let markdownwiki.path = '~/notes/markdownwiki'
    let markdownwiki.path_html = '~/notes/markdownwiki/html/'
    let markdownwiki.auto_tags = 1
    let markdownwiki.auto_diary_index = 1
    let markdownwiki.automatic_nested_syntaxes = 1
    let markdownwiki.syntax = 'markdown'
    let markdownwiki.ext = '.md'

    " a third wiki for testing https://github.com/vimwiki/vimwiki/issues/1146
    let thirdwiki = {}
    let thirdwiki.path = '/tmp/thirdwiki'
    let thirdwiki.path_html = '/tmp/thirdwiki/html/'
    let thirdwiki.auto_tags = 1
    let thirdwiki.auto_diary_index = 1
    let thirdwiki.automatic_nested_syntaxes = 1
    let thirdwiki.syntax = 'markdown'
    let thirdwiki.ext = '.md'

    " there can be many of these
    " let g:vimwiki_list = [wiki, markdownwiki, thirdwiki]
    let g:vimwiki_list = [wiki, markdownwiki, thirdwiki]

    " don't make temporary wikis based on file extensions in the list - this
    " is necessary to avoid .md files getting a filetype of vimwiki instead of
    " markdown:
    let g:vimwiki_global_ext = 0

    let g:vimwiki_folding = 'expr'
    " let g:vimwiki_folding = 'syntax'
    " let g:vimwiki_folding = 'list'

    " links concealing, etc.
    " let g:vimwiki_conceallevel = 0

    " don't obscure URLs in links
    " let g:vimwiki_url_maxsave = 0

    " make links clickable and such
    let g:vimwiki_use_mouse = 1

    " use colors in header highlighting
    let g:vimwiki_hl_headers = 1

    " change to dir of current page, theoretically
    let g:vimwiki_auto_chdir = 1

    Plugin 'vimwiki/vimwiki'

  " }}}

  call vundle#end()
  filetype plugin on
  filetype indent on

" }}}
```

**Comment by jmtd** [comment]
Hi,

> One question - I'm assuming you have g:vimwiki_auto_chdir set?

Nope; I haven't set it explicitly, and `:echo g:vimwiki_auto_chdir` reports it as an undefined variable.

> I made a cursory attempt to reproduce this with the below config and dev branch at 22dcb36; didn't come up with much.

I'll try something similar.

**Comment by jmtd** [comment]
TL;DR: this is a [taskwiki](https://github.com/tools-life/taskwiki) bug/behaviour.

I'm not sure why I didn't think of this before, but I just managed a minimal reproducer and it turns out taskwiki has to be loaded for it to happen. I'll close this here, and possibly open another over there.

Thanks!
